includes/Modules/Ads.php

```diff
@@ -126,8 +131,19 @@ protected function setup_assets() {
 					},
 				)
 			);
-
-			if ( current_user_can( Permissions::VIEW_AUTHENTICATED_DASHBOARD ) && $this->is_connected() ) {
+			// Integrator should be included if either Ads module is connected already,
+			// or we are on the Ads module setup screen.
+			if (
+				current_user_can( Permissions::VIEW_AUTHENTICATED_DASHBOARD ) &&
+				(
+					// Integrator should be included if either:
+					//
+					// The Ads module is already connected.
+					$this->is_connected() ||
+					// Or the user is on the Ads module setup screen.
+					is_admin() && 'googlesitekit-dashboard' === $this->context->input()->filter( INPUT_GET, 'page' ) && 'ads' === $this->context->input()->filter( INPUT_GET, 'slug' ) && $this->context->input()->filter( INPUT_GET, 'slug' )
```

Comment:
`slug` is duplicated on this line. Did you mean to add `reAuth` here? Please see other instances where it is used for reference.

As an aside, it's a bit verbose to repeat `$this->context->` several times. We can store the `Input` instance above to make it a bit easier to read – e.g. https://github.com/google/site-kit-wp/blob/0a9e4ac725bffd8cdf2825518d0b9463518edf4d/includes/Core/Authentication/Authentication.php#L752-L753

Also, stacking the conditions on their own line makes it easier to read as well.

Reply:
Was planning to include `reAuth` instead of slug 🤦